pkg/image/options.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package image
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// optionsRoundTripper ensures that the insecure flag is honored, and sets headers on outgoing requests
+type optionsRoundTripper struct {
+	inner       http.RoundTripper
+	httpHeaders map[string]string
+	insecure    bool
+}
+
+// RoundTrip implements http.RoundTripper
+func (ort *optionsRoundTripper) RoundTrip(in *http.Request) (*http.Response, error) {
+	// Make sure that HTTP traffic is only used if insecure is set to true
+	if !ort.insecure && in.URL != nil && in.URL.Scheme == "http" {
+		return nil, errors.New("the http protocol is not allowed")
+	}
+
+	for k, v := range ort.httpHeaders {
+		in.Header.Set(k, v)
+	}
+
+	in.Header.Set("User-Agent", "Shipwright Build")
```

Comment:
Should be make this configurable for potential upstream adopters of the framework?

Reply:
Can you check my change, I still have it hard-coded in the main.go now. If you meant to also change this, let me know.

Reply:
Thanks. I think we saw the update and it makes sense. Can be changed once we feel we need to.